src/TypeSpec.Extension/Emitter.Csharp/src/lib/confidentLevels.ts

```diff
@@ -0,0 +1,78 @@
+import {
+    InputDictionaryType,
+    InputListType,
+    InputModelType,
+    InputType,
+    isInputDictionaryType,
+    isInputListType,
+    isInputModelType
+} from "../type/inputType.js";
+
+export function getConfident(
+    type: InputType,
+    visitedModels: Set<InputModelType>
+): boolean {
+    if (isInputModelType(type)) {
+        return getModelConfident(type, visitedModels);
+    } else if (isInputListType(type)) {
+        return getListConfident(type, visitedModels);
+    } else if (isInputDictionaryType(type)) {
+        return getDictionaryConfident(type, visitedModels);
+    }
+
+    return type.IsConfident ?? true; // by default they are confident
+}
+
+function getModelConfident(
+    type: InputModelType,
+    visitedModels: Set<InputModelType>
+): boolean {
+    if (visitedModels.has(type)) return true; // this means this has been calculated before or being calculated right now, return
+    visitedModels.add(type);
+
+    if (type.IsConfident !== null) return type.IsConfident;
+
+    const isBaseLowConfident = type.BaseModel
+        ? getModelConfident(type.BaseModel, visitedModels)
+        : true;
+
+    let isModelLowConfident = isBaseLowConfident;
+    for (const prop of type.Properties) {
+        isModelLowConfident &&= getConfident(prop.Type, visitedModels);
+    }
+
+    return isModelLowConfident;
+}
+
+function getListConfident(
+    type: InputListType,
+    visitedModels: Set<InputModelType>
+): boolean {
+    return getConfident(type.ElementType, visitedModels);
+}
+
+function getDictionaryConfident(
+    type: InputDictionaryType,
+    visitedModels: Set<InputModelType>
+): boolean {
+    return (
+        getConfident(type.KeyType, visitedModels) &&
+        getConfident(type.ValueType, visitedModels)
+    );
+}
+
+// we use this method to get the confident level value of a type
+// TODO -- in the future, we could consider if we could convert the InputType and all its derived types into a concrete type instead of interfaces, so that they could contain method and method override. If so, we no longer need to do this.
+export function isConfident(type: InputType | undefined): boolean {
+    if (type === undefined) return true; // if undefined, return true
```

Comment:
If `null`, we throw error?

Reply:
This is for type, not the isConfident property.

There is some cases that when we call this, the type might be undefined (for instance, when we call this for the response), I do not want to put a if statement on the caller, therefore I put it here.